File: Android/API_Android/WeatherAppWithC/main/java/com/example/weatherappcapi/Network/WeatherAPI.kt
package com.example.weatherappcapi.Network

import com.example.weatherappcapi.JsonData.Weather
import retrofit2.http.GET
import retrofit2.http.Path

interface WeatherAPI {

    @GET("category/pmp3g/version/2/geotype/point/lon/{lon}/lat/{lat}/data.json")
    suspend fun getWeather(@Path("lon") lon: Double, @Path("lat") lat: Double): Weather
}

File: HTML5/HTML5_API/weatherAppCAPI/weatherAppCAPI/API/weatherApi.js


const fetchData = async (lat, lon) => {
    try {
      const response = await fetch(`https://opendata-download-metfcst.smhi.se/api/category/pmp3g/version/2/geotype/point/lon/${lon}/lat/${lat}/data.json`,{ 
        method: 'GET',
        cache: 'no-store'  // Lägg till det här
      });
    
      if (!response.ok) {
        throw new Error(`API request failed with status ${response.status}`);
      }

      const data = await response.json();
      return data;
    } catch (error) {
      console.error("Error fetching weather data:", error);
      throw error;
    }
  };